Update thermostat documentation in line with changes for #3298 (#1966)
OttoWinter/esphomedocs: components/climate/thermostat.rst

**If configured, at least one of** ``default_target_temperature_low`` **and** ``default_target_temperature_high``
**must be specified in the away mode configuration.**

Presets
*******

> Presets allow you to preconfigure different modes for your thermostat. These will show up in the Home Assistant

Reviewer: I think I would avoid the word "modes" to describe what presets are. I might word it this way:

_"Presets allow you to define a combination of set points, climate, fan, and swing modes that can be recalled from the front end (Home Assistant) collectively for quick, easy access. This can simplify the user experience and automations, among other things."_

UI where they can change temperature set points, fan modes, fan swing modes, and operating modes all with one
operation.

- **preset**: (*Optional*, list)

  - **name** (*Required*, string): Name of the preset. If this is one of the *standard* presets (``eco``, ``away``, 

>     ``boost``, ``comfort``, ``home``, ``sleep``, or ``activity``) it is considered a *standard_*preset. Any other

Reviewer: There's something weird with the RST formatting going on here. I think you tried to do markdown but it's RST. 😁

>     string will make the preset a *custom* preset. *Standard* and *custom* presets are functionally equivalent

Reviewer: More markdown I think. 😄 Also, there should be a comma after the word "equivalent".

    the only difference is that when switching the mode via :ref:`climate.control Action <climate-control_action>`
    you will need to use the `preset` or `custom_preset` property as appropriate. The Home Assistant 
    `climate.set_preset_mode` service treats them identically
  - **default_target_temperature_low** (*Optional*, float): The default low target temperature when switching to 
    this preset
  - **default_target_temperature_high** (*Optional*, float): The default high target temperature when switching
    to this preset.

>   - **mode** (*Optional*, climate mode): The default mode to switch the thermostat when switching to this preset

Reviewer: ```suggestion
  - **mode** (*Optional*, climate mode): The mode the thermostat should switch to when this preset is activated.
```

>     If not specified it will remain on the mode it was on prior to switching. One of:

Reviewer: ```suggestion
    If not specified, the thermostat's mode will remain unchanged when the preset is activated. One of:
```

  
    - ``heat_cool``
    - ``cool``
    - ``heat``
    - ``dry``
    - ``fan_only``
    - ``auto``
  - **fan_mode** (*Optional*, climate fan mode): The default fan mode to switch the thermostat to when switching

>     to this preset. If not specified it will remain on the fan mode it was on prior to switching. One of:

Reviewer: As above (sorry, it just reads more smoothly 😇).

    - ``on``
    - ``off``
    - ``auto``
    - ``low``
    - ``medium``
    - ``high``
    - ``middle``
    - ``focus``
    - ``diffuse``
  - **swing_mode** (*Optional*, climate swing mode): The default fan swing mode to switch the thermostat to when

>     switching to this preset. If not specified it will remain on the fan mode it was on prior to switching. One of:

Reviewer: One more time (sorry, it just reads more smoothly 😇). haha

**If configured, at least one of** ``default_target_temperature_low`` **and** ``default_target_temperature_high``
**must be specified in the away mode configuration.**

Presets
*******

> Presets allow you to preconfigure different modes for your thermostat. These will show up in the HomeAssistant

Reviewer: "Home Assistant" is two words. 🙂

Author: Cheers!

Not sure if you're aware but you can also leave suggestions by leaving a line-specific comment with a multi-line code block with the language of "suggestion" (ie. ```suggestion...). Then the Github UI let's the owner just click accept the suggestions and commit them. Not suitable for all issues on a PR, but great for typos!

    If not specified it will remain on the mode it was on prior to switching. One of:
  
    - ``heat_cool``
    - ``cool``
    - ``heat``
    - ``dry``
    - ``fan_only``
    - ``auto``
  - **fan_mode** (*Optional*, climate fan mode): The default fan mode to switch the thermostat to when switching
    to this preset. If not specified it will remain on the fan mode it was on prior to switching. One of:

    - ``on``
    - ``off``
    - ``auto``
    - ``low``
    - ``medium``
    - ``high``
    - ``middle``
    - ``focus``
    - ``diffuse``
  - **swing_mode** (*Optional*, climate swing mode): The default fan swing mode to switch the thermostat to when
    switching to this preset. If not specified it will remain on the fan mode it was on prior to switching. One of:
    
    - ``off``
    - ``both``
    - ``horizontal``
    - ``vertical``

.. code-block:: yaml

    # Example climate controller with presets
    climate:
      - platform: thermostat
        name: "Thermostat with Presets"
        preset:
          sleep:
            default_target_temperature_low: 17
            default_target_temperature_high: 26

>             fan_mode: LOw

Reviewer: Weird casing?

            swing_mode: OFF
          boost:
            default_target_temperature_low: 21
            default_target_temperature_high: 23
            fan_mode: HIGH
            mode: HEAT_COOL

- **preset_change**: (*Optional*, :ref:`Action <config-action>`): The action to call when the preset is changed. This

>   will be called either when a user changes the mode through the HomeAssistant UI or through a call to ``climate.control``

Reviewer: "Home Assistant" is two words. 🙂

.. code-block:: yaml

    # Example climate controller with presets actions
    climate:
      - platform: thermostat
        name: "Thermostat with Presets Actions"
        preset:
          sleep:
            default_target_temperature_low: 17
            default_target_temperature_high: 26

>             fan_mode: LOw

Reviewer: Weird casing again -- I think copy/paste got you. 🙂
